WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailCardBuilder.kt

```diff
@@ -615,8 +620,13 @@ class ProductDetailCardBuilder(
         }
     }
 
-    private fun Product.addons(): ProductProperty? =
-        takeIf { addons.isNotEmpty() && AppPrefs.isProductAddonsEnabled }?.let {
+    private suspend fun Product.addons(): ProductProperty? =
+        takeIf { product ->
+            addonsStore.observeProductSpecificAddons(
+                siteRemoteId = selectedSite.get().siteId,
+                productRemoteId = product.remoteId
+            ).firstOrNull().isNullOrEmpty().not() && AppPrefs.isProductAddonsEnabled
+        }?.let {
```

Comment:
I was wondering if we should leave this kind of operation to the `AddonRepository` to perform and avoid attaching the FluxC `WCAddonStore` directly to the card builder, something like this:
```suggestion
    private suspend fun Product.addons(): ProductProperty? =
        takeIf { addonsRepository.containsAddonsWith(it.remoteId) && AppPrefs.isProductAddonsEnabled }?.let {
```

And at the `AddonRepository`:
```
    suspend fun containsAddons(productID: Long) : Boolean =
        addonsStore.observeProductSpecificAddons(
            siteRemoteId = selectedSite.get().siteId,
            productRemoteId = productID
        ).firstOrNull().isNullOrEmpty().not()
```

This way we can do unit tests for this operation inside `AddonRepositoryTest` and make it easier to read the conditional rule defined here. WDYT?

Reply:
I agree! Especially I like that we don't have to introduce `SelectedSite` in many other places. Also improves encapsulation. Added in b53addba1c7a09730e787b6dcba67d877fe33e75 with a little rename.